letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -398,66 +443,71 @@ def is_name_vhost(self, target_addr):
         # search for NameVirtualHost directive for ip_addr
         # note ip_addr can be FQDN although Apache does not recommend it
         return (self.version >= (2, 4) or
-                self.parser.find_dir(
-                    parser.case_i("NameVirtualHost"),
-                    parser.case_i(str(target_addr))))
+                self.parser.find_dir("NameVirtualHost", str(target_addr)))
 
     def add_name_vhost(self, addr):
         """Adds NameVirtualHost directive for given address.
 
-        :param str addr: Address that will be added as NameVirtualHost directive
+        :param addr: Address that will be added as NameVirtualHost directive
+        :type addr: :class:~letsencrypt_apache.obj.Addr
 
         """
-        path = self.parser.add_dir_to_ifmodssl(
-            parser.get_aug_path(
-                self.parser.loc["name"]), "NameVirtualHost", str(addr))
+        loc = parser.get_aug_path(self.parser.loc["name"])
+
+        if addr.get_port() == "443":
```

Comment:
Throughout Apache... ports are strings unless otherwise specified.  (It is in the documentation for Addr() ) All arguments/values from configuration files are strings and the port's integer value isn't used anywhere.  I did attempt to convert everything over to using integer ports at one point, but I was constantly typecasting. :/